### PR TITLE
Separating GitHub codeowners and aws access #5908

### PR DESCRIPTION
**Pull request by georgepstaylor**
## A reference to the issue / Description of it

See #5908 

## How does this PR fix the problem?

Allows specifying a top level array containing github_slugs to be considered as `codeowners`. If you have not provided a `codeowners` key, then the existing method of considering all `github_slugs` with member access as codeowners is still used.

## How has this been tested?

Please describe the tests that you ran and provide instructions to reproduce.

I have tested the bash locally using an example `${environment_json_dir}/${application_name}.json` file. 

## Deployment Plan / Instructions

Will this deployment impact the platform and / or services on it?

{Please write here}

## Checklist (check `x` in `[ ]` of list items)

- [X] I have performed a self-review of my own code
- [x] All checks have passed
- [ ] I have made corresponding changes to the documentation
- [ ] Plan and discussed how it should be deployed to PROD (If needed)

## Additional comments (if any)

{Please write here}

### CODE REVIEW

**Comment by dms1981** [comment]
Hi @georgepstaylor - could you provide an example as to what the `environments/*.json` would look like with this approach?

**Comment by georgepstaylor** [comment]
> Hi @georgepstaylor - could you provide an example as to what the `environments/*.json` would look like with this approach?

Of course! Full example for `delus-core.json`

```json
{
  "account-type": "member",
  "codeowners" : ["hmpps-migration"],
  "environments": [
    {
      "name": "development",
      "access": [
        {
          "github_slug": "hmpps-migration",
          "level": "sandbox",
          "nuke": "exclude"
        },
        {
          "github_slug": "hmpps-dba",
          "level": "sandbox",
          "nuke": "exclude"
        },
        {
          "github_slug": "unilink",
          "level": "instance-management"
        }
      ]
    },
    {
      "name": "test",
      "access": [
        {
          "github_slug": "hmpps-migration",
          "level": "developer"
        }
      ]
    }
  ],
  "tags": {
    "application": "delius",
    "business-unit": "HMPPS",
    "infrastructure-support": "probation-webops@digital.justice.gov.uk",
    "owner": "probation-webops@digital.justice.gov.uk"
  },
  "github-oidc-team-repositories": [
    "ministryofjustice/hmpps-delius-operational-automation",
    "ministryofjustice/hmpps-openldap-container"
  ]
}
```

**Comment by dms1981** [comment]
Ahhh OK! @georgepstaylor I had a candidate in mind that looked a little different:
```
      "access": [
        {
          "github_slug": "team-1",
          "level": "developer"
        },
        {
          "github_slug": "team-2",
          "level": "instance-management"
          "codeowners": "exclude"
        }
      ]
```
With a change to the `jq` that populates the `github_slugs` variable to look like this:
```
github_slugs=$(jq -r '.environments[].access[] | select(.codeowners != "exclude") | .github_slug | "@ministryofjustice/" + .' ${environment_json_dir}/${application_name}.json | sort | uniq | tr '\n' ' ')
```
But since this isn't my ticket, I'll leave this to the person who picked up the work to make the decision :)

**Comment by georgepstaylor** [comment]
@dms1981 Yeah that actually looks very similar to the approach I originally suggested in the issue. I hit a problem with it in my testing - if a github team appears in more than one environment you'd have to have the `"codeowners": "exclude"` duplicated in each of the `access` lists. Then that got me thinking that `codeowners` isn't really an environment level control as it relates to code which is on main and thus environment agnostic

**Review by SteveLinden — APPROVED** [review]

**Pull request merged**
